projects/plugins/boost/app/lib/Features_Status.php

```diff
@@ -0,0 +1,35 @@
+<?php
+/**
+ * Class to check the status of Boost features.
+ *
+ * @package automattic/jetpack-boost
+ */
+
+namespace Automattic\Jetpack_Boost\Lib;
+
+use Automattic\Jetpack_Boost\Lib\Critical_CSS\Critical_CSS_State;
+
+/**
+ * Class that handles checking the status of Jetpack Boost features.
+ */
+class Features_Status {
+
+	const GOOD = 'good';
+	const BAD  = 'bad';
+
+	public static function get_total_problems() {
+		$status = self::get_cloud_css_state_status();
+		return $status === self::GOOD ? 0 : 1;
```

Comment:
@thingalon suggested this, and I think it would be elegant as well. Rather than manually checking everything within this method, we could use a filter to get the count of problems. This would look something like this:

```
return apply_filter( 'jetpack_boost_total_problem_count', 0 );
```

And then, each active optimization feature or anything else can hook into this and increase the count. This has the advantage of letting the features worry about determining how they identify problems.

Reply:
I'm having a bit of trouble with this one. The `apply_filters` code would get executed way before features are initialised:
https://github.com/Automattic/jetpack/blob/4d5f0b4814f6d0d3e48402b5e3492a2e583d8fc7/projects/plugins/boost/app/admin/class-admin.php#L62-L73
I added a health check method to [Cloud_CSS](https://github.com/Automattic/jetpack/blob/4d5f0b4814f6d0d3e48402b5e3492a2e583d8fc7/projects/plugins/boost/app/features/optimizations/cloud-css/Cloud_CSS.php), but I'm not sure where to hook it so it gets initialised before the menu does.

I imagine adding the bubble with JS is out of the question 😅

Reply:
> I'm having a bit of trouble with this one. The apply_filters code would get executed way before features are initialised

🤔 Oh. That does make it annoying.

`Admin_Menu::add_menu` has to be called before the `admin_menu` hook. All of our optimizations get initialized by the `init` hook.

We could move this call to `Admin_Menu::add_menu` to a late-firing `init` hook, or an early-firing `admin_menu` hook.

Reply:
Thank you for the pointers @thingalon!

I've made some changes to the way checks are handled here 5fb2c30dad44ce4fb4006e92ca55324b1c782f5e.

Those include moving the `Admin_Menu::add_menu` to the `admin_menu` hook. I went through the code and there shouldn't be any problems with this move.

I also scrapped the whole Features_Status class in favour of methods inside the feature itself. Though I'm not a 100% happy with having to hook the method from within each feature. I couldn't find the right place to add a loop that goes through all features, checks if they have the "health check" method and updating the problem count if necessary.